java-sdk/src/main/java/com/netflix/conductor/sdk/executor/WorkflowExecutor.java

```diff
@@ -0,0 +1,370 @@
+            }
+            healthCheckExecutor.shutdownNow();
+
+            ses.scheduleAtFixedRate(()->{
+
+                runningWorkflows.entrySet().forEach(e -> {
+                    String workflowId = e.getKey();
+                    CountDownLatch latch = e.getValue();
+                    Workflow workflow = workflowClient.getWorkflow(workflowId, false);
+                    if(workflow.getStatus().isTerminal()) {
+                        latch.countDown();
+                    }
+                });
+            }, 100, 100, TimeUnit.MILLISECONDS);
+
+        } catch (IOException e) {
+            throw new Error(e);
+        }
+    }
+
+    /**
+     * Finds any worker implementation and starts polling for tasks
+     * @param basePackage list of packages - comma separated - to scan for annotated worker implementation
+     */
+    public void initWorkers(String basePackage) {
+        scanWorkers(basePackage);
+        startPolling();
+    }
+
+
+
+    private void startPolling() {
+
+        List<Worker> executors = new ArrayList<>();
+        workerExecutors.entrySet().forEach( e-> {
+            String taskName = e.getKey();
+            Method method = e.getValue();
+            Object obj = workerClassObjs.get(taskName);
+            WorkerExecutor executor = new WorkerExecutor(taskName, method, obj);
+            executors.add(executor);
+        });
+        if(executors.isEmpty()) {
+            return;
+        }
+
+        taskRunner = new TaskRunnerConfigurer.Builder(taskClient, executors)
+                .withThreadCount(executors.size())
+                .build();
+        taskRunner.init();
+    }
+
+
+
+    private synchronized void installAndStartServer(String repositoryURL, String localServerPort) throws IOException {
+
+        if(serverProcess != null) {
+            return;
+        }
+
+        String configFile = WorkflowExecutor.class.getResource("/test-server.properties").getFile();
+        String tempDir = System.getProperty("java.io.tmpdir");
+        Path serverFile = Paths.get(tempDir, "conductor-server.jar");
+        if(!Files.exists(serverFile)) {
+            Files.copy(new URL(repositoryURL).openStream(), serverFile);
+        }
+        serverProcess = Runtime.getRuntime().exec("java -Dserver.port=" + localServerPort + " -DCONDUCTOR_CONFIG_FILE=" + configFile + " -jar " + serverFile.toString());
+        InputStream in = serverProcess.getInputStream();
+        InputStream err = serverProcess.getErrorStream();
+        BufferedReader error = new BufferedReader(new InputStreamReader(serverProcess.getErrorStream()));
+        BufferedReader op = new BufferedReader(new InputStreamReader(serverProcess.getInputStream()));
+
+        Executors.newSingleThreadScheduledExecutor().execute(()->{
+
+            String line = null;
+            while(true) {
+                try {
+                    if (!((line=error.readLine()) != null)) break;
```

Comment:
what is the purpose of this? So it will break out the loop when there is any error line in server log, but server still keeps running

Reply:
I've simplified this - it breaks out only if its null. And readLine should wait.